src/integrationTest/java/org/opensearch/security/SnapshotSteps.java

```diff
@@ -58,13 +60,21 @@ public CreateSnapshotResponse createSnapshot(String repositoryName, String snaps
         return snapshotClient.create(createSnapshotRequest, DEFAULT);
     }
 
-    public void waitForSnapshotCreation(String repositoryName, String snapshotName) {
+    public int waitForSnapshotCreation(String repositoryName, String snapshotName) {
+        AtomicInteger count = new AtomicInteger();
         GetSnapshotsRequest getSnapshotsRequest = new GetSnapshotsRequest(repositoryName, new String[] { snapshotName });
-        Awaitility.await().alias("wait for snapshot creation").ignoreExceptions().until(() -> {
-            GetSnapshotsResponse snapshotsResponse = snapshotClient.get(getSnapshotsRequest, DEFAULT);
-            SnapshotInfo snapshotInfo = snapshotsResponse.getSnapshots().get(0);
-            return SnapshotState.SUCCESS.equals(snapshotInfo.state());
-        });
+        Awaitility.await()
+            .pollDelay(250, TimeUnit.MILLISECONDS)
```

Comment:
will these values hold or be stable?

Reply:
I put this part in (pollDelay()) on earlier iteration of fixing tests dealing with snapshot creation. I found that introducing a little delay before performing HTTP check call is beneficial. At this point it shouldn't really matter, I refactored the method to return a number of calls performed, so after snapshot is verified it doesn't matter if state was checked once or couple times. Asserts check returned number of calls from this method against audit messages logged.
I could remove it and check again does it passes ci correctly without these changes (it should). The default poll delay and poll interval is 100 milliseconds.
Let me know what do you think.

Reply:
If the tests seem to be stable with this delay, I don't mind keeping it. I would like to be sure that this is not bound to change inconsistently in future as it is a magic number being introduced here.

Reply:
It will be consistent thanks to the number of calls returned.